torch_np/_detail/_flips.py

```diff
@@ -0,0 +1,52 @@
+"""Implementations of flip-based routines and related animals. 
+"""
+
+import torch
+
+from . import _scalar_types, _util
+
+
+def flip(m_tensor, axis=None):
+    # XXX: semantic difference: np.flip returns a view, torch.flip copies
+    if axis is None:
+        axis = tuple(range(m_tensor.ndim))
+    else:
+        axis = _util.normalize_axis_tuple(axis, m_tensor.ndim)
+    return torch.flip(m_tensor, axis)
+
+
+def flipud(m_tensor):
+    return torch.flipud(m_tensor)
+
+
+def fliplr(m_tensor):
+    return torch.fliplr(m_tensor)
+
+
+def rot90(m_tensor, k=1, axes=(0, 1)):
+    axes = _util.normalize_axis_tuple(axes, m_tensor.ndim)
+    return torch.rot90(m_tensor, k, axes)
+
+
+def swapaxes(tensor, axis1, axis2):
+    return torch.swapaxes(tensor, axis1, axis2)
+
+
+# https://github.com/numpy/numpy/blob/v1.24.0/numpy/core/numeric.py#L1259
+def rollaxis(tensor, axis, start=0):
+    n = tensor.ndim
+    axis = _util.normalize_axis_index(axis, n)
+    if start < 0:
+        start += n
+    msg = "'%s' arg requires %d <= %s < %d, but %d was passed in"
+    if not (0 <= start < n + 1):
+        raise _util.AxisError(msg % ("start", -n, "start", n + 1, start))
```

Comment:
Two small things.
- It'd be good to upgrade these to f-strings
- It'd be good to have a `_util.check` utility. See for example https://github.com/pytorch/pytorch/blob/8c26ed5f5e36bf0692b381b16fe1b9efdba16185/torch/_prims_common/__init__.py#L1556-L1566

Reply:
Why do you people do not like the old-style string formatting :-). Nevermind :-).

On a more serious note, this whole function is a weird one out, retained in numpy for backwards compatibility:
https://stackoverflow.com/questions/29891583/reason-why-numpy-rollaxis-is-so-confusing

I'd much rather keep it as is with a permalink to the source and a link to this SO answer by the author of the improved numpy function.

Why implementing it at all. Well, maintained numpy consumers have predominantly moved to `moveaxis`. IIRC when we started this project, there was a talk about less well maintained user bases.
So I'd really prefer to just clone it from numpy with minimal mods and forget :-).

Reply:
sure, let's leave it.